### PR TITLE
Add transfer learning wrapper

### PR DESCRIPTION
**Pull request by semibran**
Allows us to sub in different models from both `keras.applications` and `tensorflow_hub` just by changing config

### CODE REVIEW

File: configs/transfer_learning.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from models.inception_v3 import InceptionV3
+
+# NOTE: unfreeze breakpoints and chosen model are generally codependent
+TRANSFER_LEARNING_BASE_MODEL = InceptionV3
+FINE_TUNING_UNFREEZE_BREAKPOINTS = ('mixed8', 'mixed6', 'mixed4')
```
Comment (semibran) [inline configs/transfer_learning.py:5]: edit base model and breakpoints here

File: models/tensorflow_hub.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+"""
+Details layer-based transfer learning models from TensorFlow Hub.
+Note that using these models may require configuring specific image sizes in `configs/model.py`.
+"""
+
+from core.transfer_learning import LayerBasedTransferLearningModel
+
+
+class InceptionV3(LayerBasedTransferLearningModel):
+    handle = 'https://tfhub.dev/google/imagenet/inception_v3/feature_vector/5'
```
Comment (jonathan-paugh) [inline models/tensorflow_hub.py:10]: Was this the one used for 299, 92% or was that V2?

**Review by jonathan-paugh — APPROVED** [review]
🍉

File: configs/transfer_learning.py (hunk 1 above)
Comment (jonathan-paugh) [reply configs/transfer_learning.py:5]: Is there docs we can use to find which breakpoints for a given model?

File: models/tensorflow_hub.py (hunk 2 above)
Comment (semibran) [reply models/tensorflow_hub.py:10]: Yep that's the one

File: configs/transfer_learning.py (hunk 1 above)
Comment (semibran) [reply configs/transfer_learning.py:5]: No don't think so unfortunately. I got these values by flipping through the layers printed by `base_model.summary()`; for InceptionV3 there are 10 mixed layers to close out each block so this just tells it to gradually unfreeze 60% of the feature extractor layers over 3 fine tuning breakpoints

**Pull request merged**